src/Compilers/Server/VBCSCompiler/MetadataCache.cs

```diff
@@ -15,8 +15,8 @@ namespace Microsoft.CodeAnalysis.CompilerServer
 {
     internal class MetadataAndSymbolCache
     {
-        // Store 100 entries -- arbitrary number
-        private const int CacheSize = 100;
+        // Store 500 entries -- Out of ~8.7M projects, only about 4,000 had more than 500 references
```

Comment:
This is really gonna kill my story telling abilities. Discussing how this number was chosen by a highly scientific process that in reality is just a number that made the least people scream was one of my favorites. Now I have to say it has some real logic behind it.

Reply:
But it does involve suggesting that we make it over 9000, so there's that